backend/src/router.js

```diff
@@ -168,5 +174,34 @@ router.get("/openai/model", (req, res, next) => {
   res.send(req.session.gptModel);
 });
 
+router.post("/phase/current", (req, res, next) => {
+  const phase = req.body?.phase;
+  if (phase) {
+    req.session.phases = switchCurrentPhase(req.session, phase);
+    res.status(200).send("Phase set to " + req.session.phases);
+  } else {
+    res.status(400).send("Missing phase");
+  }
+});
+
+router.get("/phase/current", (req, res, next) => {
+  res.send(getCurrentPhase(req.session));
+});
+
+router.get("/phase/completed", (req, res, next) => {
+  res.send(getCompletedPhases(req.session));
+});
+
+router.post("/phase/completed", (req, res, next) => {
+  const phase = req.body?.phase;
+  if (phase) {
+    req.session.phases = setPhaseCompleted(req.session, phase);
+    res.send("Phase complete status set");
+  } else {
+    res.statusCode = 400;
+    res.send("Missing phase");
+  }
+});
```

Comment:
just keep the `router.get("/phase/completed", (req, res, next) => {` endpoint, so the frontend can know which phases the user can switch to. The backend is in charge of completing phases, and the frontend will send the current phase up with the chat message.